fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -210,7 +210,7 @@
     protected final RecordMetaDataProvider metaDataProvider;
 
     @Nullable
-    protected MutableRecordStoreState recordStoreState;
+    protected AtomicReference<MutableRecordStoreState> recordStoreState = new AtomicReference<>();
```

Comment:
I think `recordStoreStateRef` instead of `recordStoreState` to underscore it's a reference. This also let's one cache `MutableRecordStoreState recordStoreState = recordStoreStateRef.get()` in functions. This should also mark the member as final, as changing the reference itself (rather than changing the thing the reference points to) should be illegal.